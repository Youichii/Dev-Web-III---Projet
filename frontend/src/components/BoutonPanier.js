import PropTypes from 'prop-types'

const BoutonPanier = ({ className, id_div, id_elem, name, value, onClick }) => {
    return (
        <div className={className} id={id_div}>
            <input id={id_elem} name={name} type="button" value={value} onClick={onClick} />
        </div>
    )

}

/*BoutonPanier.defaultProps = {
    color: 'steelblue',
    text : 'Hello'
}*/

BoutonPanier.propTypes = {
    className : PropTypes.string,
    id_div : PropTypes.string,
    id_elem : PropTypes.string,
    name : PropTypes.string,
    value : PropTypes.string,
    onClick : PropTypes.func,
}

export default BoutonPanier

