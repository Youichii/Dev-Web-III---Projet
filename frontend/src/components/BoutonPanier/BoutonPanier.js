import PropTypes from 'prop-types'

const BoutonPanier = ({ className, id_div, id_elem, name, value, onClick }) => {
    return (
        <div data-testid="div_bouton_panier" className={className} id={id_div}>
            <input data-testid="input_bouton_panier" id={id_elem} name={name} type="button" value={value} onClick={onClick} />
        </div>
    )
}

/*BoutonPanier.defaultProps = {
    color: 'steelblue',
    text : 'Hello'
}*/

BoutonPanier.propTypes = {
    className : PropTypes.string,
    id_div : PropTypes.string,
    id_elem : PropTypes.string,
    name : PropTypes.string,
    value : PropTypes.string,
    onClick : PropTypes.func,
}

export default BoutonPanier
