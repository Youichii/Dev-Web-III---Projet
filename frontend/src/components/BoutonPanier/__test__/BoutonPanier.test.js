import React from 'react';
import ReactDOM from 'react-dom';
import BoutonPanier from '../BoutonPanier';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing BoutonPanier", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BoutonPanier></BoutonPanier>, div)
})

it("render value text correctly BoutonPanier", () => {
    const {getByTestId} = render(<BoutonPanier className="i_bouton_annuler2" id_div="elem_bouton_annuler2" id_elem="bouton_annuler2" name="bout_annuler2" value="Retour"></BoutonPanier>);
    expect(getByTestId('input_bouton_panier')).toHaveAttribute('value', 'Retour');
})

it("render id correctly BoutonPanier", () => {
    const {getByTestId} = render(<BoutonPanier className="i_bouton_annuler2" id_div="elem_bouton_annuler2" id_elem="bouton_annuler2" name="bout_annuler2" value="Retour"></BoutonPanier>);
    expect(getByTestId('div_bouton_panier')).toHaveAttribute('id', 'elem_bouton_annuler2');
})

it("render type input correctly BoutonPanier", () => {
    const {getByTestId} = render(<BoutonPanier className="i_bouton_annuler2" id_div="elem_bouton_annuler2" id_elem="bouton_annuler2" name="bout_annuler2" value="Retour"></BoutonPanier>);
    expect(getByTestId('input_bouton_panier')).toHaveAttribute('type', 'button');
})

it("matches snapshot 1 BoutonPanier", () => {
    const tree = renderer.create(<BoutonPanier className="i_bouton_annuler2" id_div="elem_bouton_annuler2" id_elem="bouton_annuler2" name="bout_annuler2" value="Retour"></BoutonPanier>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot(); 
}) 

