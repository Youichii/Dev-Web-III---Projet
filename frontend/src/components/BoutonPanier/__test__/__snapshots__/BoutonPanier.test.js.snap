// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 BoutonPanier 1`] = `
<div
  className="i_bouton_annuler2"
  data-testid="div_bouton_panier"
  id="elem_bouton_annuler2"
>
  <input
    data-testid="input_bouton_panier"
    id="bouton_annuler2"
    name="bout_annuler2"
    type="button"
    value="Retour"
  />
</div>
`;

exports[`matches snapshot 2 BoutonPanier 1`] = `
<div
  className="i_bouton_annuler2"
  data-testid="div_bouton_panier"
  id="elem_bouton_annuler2"
>
  <input
    data-testid="input_bouton_panier"
    id="bouton_annuler2"
    name="bout_annuler2"
    type="button"
    value="Retour"
  />
</div>
`;
